File: go/jogo.go
package main

import (
	"fmt"
	"math/rand/v2"
)

func main() {
	numero_escolhido := rand.IntN(100) + 1
	var n, tentativas int = 0, 0

	for tentativas < 7 {
		fmt.Printf("Digite seu palpite (1 a 100): ")
		fmt.Scanln(&n)
		tentativas++

		if n == numero_escolhido {
			fmt.Printf("\nParabéns! Acertaste em cheio em apenas %v tentativas\n", tentativas)
			break
		} else {
			if n < numero_escolhido {
				fmt.Println("Baixo!")
			} else {
				fmt.Println("Alto!")
			}
		}

		if tentativas == 7 && n != numero_escolhido {
			fmt.Printf("\nO número era: %v. Fim de jogo!!\n", numero_escolhido)
		}
	}
}

File: c/jogo.c
#include <stdio.h>
#include <stdlib.h>
int main(){

    int numero_escolhido = rand() % 101;
    int tentativas = 0;
    int n;

    while (tentativas < 7)
    {
        printf("Digite seu palpite (1 a 100): ");
        scanf("%d",&n);
        tentativas++;
        if(n == numero_escolhido){
            printf("\nParabéns acertaste em cheio em apenas %d tentativas\n",tentativas);
            break;
        }else{
            if(n<numero_escolhido)
                printf("Baixo\n");
            else
                printf("Alto\n");
        }
        if(tentativas == 7 && n != numero_escolhido){
            printf("O número era: %d. Fim de jogo!!",numero_escolhido);
        }
    }
    
    


    return 0;
}